app/scenes/channel/channel_post_list/channel_post_list.js

```diff
@@ -16,12 +16,17 @@ export default class ChannelPostList extends PureComponent {
         posts: PropTypes.array.isRequired
     };
 
+    state = {
+        lastViewedAt: this.props.myMember.last_viewed_at
+    };
+
     componentDidMount() {
         this.props.actions.loadPostsIfNecessary(this.props.channel);
     }
 
     componentWillReceiveProps(nextProps) {
         if (this.props.channel.id !== nextProps.channel.id) {
+            this.setState({lastViewedAt: nextProps.myMember.last_viewed_at});
```

Comment:
There's a known issue when you leave the channel that myMember is not set, maybe we should handle that in this PR, I know the prop is set as required but if you leave the channel the member will be null